xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/match/BlockNavigator.java

```diff
@@ -239,14 +247,21 @@ public BlockNavigator(BlockMatcher matcher)
                     nextBlock = block.getParent();
                     break;
                 // DESCENDANT
-                case CHILD:
-                    List<Block> children = block.getChildren();
-                    if (!children.isEmpty()) {
-                        nextBlock = children.get(0);
-                        axes = Axes.FOLLOWING_SIBLING;
-                        if (this.matcher.match(nextBlock)) {
-                            return (T) nextBlock;
+                /* In order to reduce number of lines from 9 to 5,
+                   all variables with the exception of axes were removed from the code.
+                        nextBlock = block.getChildren().get(0)
+                        children = block.getChildren()
+                   The function returnFollowingAxes executes takes the place of the previously used code, shown below
+                        if (!block.getChildren().isEmpty()) {
+                            axes = Axes.FOLLOWING_SIBLING;
                         }
+                   The function returns Axes.FOLLOWING_SIBLING if the condition is true, and if false returns
+                   the current axes.
+                 */
+                case CHILD:
+                    axes = returnFollowingAxes(block, currentAxes);
```

Comment:
I don't like the fact that this block become inconsistent with others (return a block instead of setting it).

Reply:
What do you suggest I do?

Reply:
Not idea but right now I still prefer the previous version.